src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpCheck.java

```diff
@@ -303,38 +303,41 @@ public void beginTree(DetailAST rootAST) {
     }
 
     /**
-     * Recursive method that finds the matches.
+     * Finds the matches in the file.
      *
-     * @noinspection TailRecursion
-     * @noinspectionreason TailRecursion - until issue #14814
      */
     // suppress deprecation until https://github.com/checkstyle/checkstyle/issues/11166
     @SuppressWarnings("deprecation")
     private void findMatch() {
-        final boolean foundMatch = matcher.find();
-        if (foundMatch) {
-            final FileText text = getFileContents().getText();
-            final LineColumn start = text.lineColumn(matcher.start());
-            final int startLine = start.getLine();
-
-            final boolean ignore = isIgnore(startLine, text, start);
-
-            if (!ignore) {
-                matchCount++;
-                if (illegalPattern || checkForDuplicates
-                        && matchCount - 1 > duplicateLimit) {
-                    errorCount++;
-                    logMessage(startLine);
+        boolean valid = true;
+        while (valid) {
+            final boolean foundMatch = matcher.find();
+            if (foundMatch) {
+                final FileText text = getFileContents().getText();
+                final LineColumn start = text.lineColumn(matcher.start());
+                final int startLine = start.getLine();
+
+                final boolean ignore = isIgnore(startLine, text, start);
+                if (!ignore) {
+                    matchCount++;
```

Comment:
This class field is used within the current method only, and it is reset every time this method is called. It made sense to have it as a class field due to recursion, but now it doesn't. Could you replace it with a local variable?

Reply:
done. should i remove ```matchCount``` class field altogther as it has no use now ?

Reply:
We can get rid of global `matcher` and `errorCount` as well. It is ok to extend dependent methods with additional parameters.